vavr/src/test/java/org/jdbi/v3/vavr/TestVavrTupleRowMapperFactoryWithDB.java

```diff
@@ -0,0 +1,183 @@
+        handle.registerRowMapper(new SomethingMapper());
+
+        Tuple1<Something> result = handle
+                .createQuery("select id, name from something where id = 1")
+                .mapTo(new GenericType<Tuple1<Something>>() {})
+                .findOnly();
+
+        assertThat(result._1).isEqualTo(new Something(1, "eric"));
+    }
+
+    @Test
+    public void testMapTuple2UsingRegisteredRowMappers_shouldSucceed() throws SQLException {
+        Handle handle = dbRule.getSharedHandle();
+        handle.registerRowMapper(new SomethingMapper());
+        handle.registerRowMapper(SomethingValues.class,
+                (rs, ctx) -> new SomethingValues(rs.getInt("integerValue"),
+                        rs.getInt("intValue")));
+
+        Tuple2<Something, SomethingValues> result = handle
+                .createQuery("select * from something where id = 2")
+                .mapTo(new GenericType<Tuple2<Something, SomethingValues>>() {})
+                .findOnly();
+
+        assertThat(result._1).isEqualTo(new Something(2, "brian"));
+        assertThat(result._2).isEqualTo(new SomethingValues(101, 102));
+    }
+
+    @Test
+    public void testMapTuple1HavingOnlyOneRowMapper_shouldFail() throws SQLException {
+        assertThatThrownBy(() -> dbRule.getSharedHandle()
+                .createQuery("select * from something where id = 1")
+                .mapTo(new GenericType<Tuple2<Something, SomethingValues>>() {})
+                .findOnly()
+        ).isInstanceOf(NoSuchMapperException.class);
+    }
+
+    @Test
+    public void testMapTuple3WithExtraSpecifiedColumn_shouldSucceed() throws SQLException {
+        Handle handle = dbRule.getSharedHandle();
+        handle.registerRowMapper(new SomethingMapper());
+        handle.configure(TupleMappers.class, c ->
+                c.setColumn(2, "integerValue").setColumn(3, "intValue"));
+
+        Tuple3<Something, Integer, Integer> result = handle
+                .createQuery("select * from something where id = 1")
+                .mapTo(new GenericType<Tuple3<Something, Integer, Integer>>() {})
+                .findOnly();
+
+        assertThat(result._1).isEqualTo(new Something(1, "eric"));
+        assertThat(result._2).isEqualTo(99);
+        assertThat(result._3).isEqualTo(100);
+    }
+
+    @Test
+    public void testMapTuple3WithoutSpecifiedColumn_shouldFail() throws SQLException {
+        Handle handle = dbRule.getSharedHandle();
+        handle.registerRowMapper(new SomethingMapper());
+
+        Tuple3<Integer, Something, Integer> result = handle
+                .createQuery("select * from something where id = 1")
+                .mapTo(new GenericType<Tuple3<Integer, Something, Integer>>() {})
+                .findOnly();
+
+        assertThat(result._1).isEqualTo(1);
+        assertThat(result._2).isEqualTo(new Something(1, "eric"));
+
+        // TODO should this fail?
+        assertThat(result._3).isNotEqualTo(100);
+    }
```

Comment:
Test name says test should fail, but nothing in the test expects an exception.

I suspect if you pull the latest changes in master into your branch, this test will start throwing an exception as expected.

Reply:
Thats what I was anticipating. Will fix all of this next week

Reply:
One question tough: I would expect a `NoSuchMapperException` in this case. `ResultBearing#mapTo` is throwing an UnsupportedOperationException.

Should i assert the unsupported or is this subject to change?

Reply:
It should be `NoSuchMapperException`. Feel free to change and adjust assertions as needed.